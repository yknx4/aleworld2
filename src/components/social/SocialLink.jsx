import React,{Component, PropTypes} from 'react';

export default class SocialLink extends Component {
  render () {
    const {icon, url, target} = this.props;
    return <a href={url} target={target}><i className={`icon icon-sm ti-${icon}`} /></a>;
  }
}

SocialLink.propTypes = {
  icon: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  target: PropTypes.string
};

SocialLink.defaultProps = {
  target: 'blank'
};
