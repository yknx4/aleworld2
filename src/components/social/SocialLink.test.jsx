import React from 'react';
import { shallow, mount } from 'enzyme';
import SocialLink from './SocialLink';

describe('SocialLink', () => {
  let shallowComponent;
  let mountedComponent;

  beforeEach(() => {
    const component = <SocialLink icon='icon' url='#'/>;
    shallowComponent = shallow(component);
    mountedComponent = mount(component);
  });

  it('should render without throwing an error', function() {
    expect(shallowComponent.contains(<a href='#' target='blank'><i className={`icon icon-sm ti-icon`} /></a>)).toBe(true);
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should be selectable by tag "a"', function() {
    expect(shallowComponent.is('a')).toBe(true);
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should mount in a full DOM', function() {
    expect(mountedComponent.find('.icon').length).toBe(1);
    expect(mountedComponent).toMatchSnapshot();
  });

});
