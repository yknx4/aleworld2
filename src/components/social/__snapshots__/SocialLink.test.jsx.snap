// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink should be selectable by tag "a" 1`] = `
<a
  href="#"
  target="blank"
>
  <i
    className="icon icon-sm ti-icon"
  />
</a>
`;

exports[`SocialLink should mount in a full DOM 1`] = `
<SocialLink
  icon="icon"
  target="blank"
  url="#"
>
  <a
    href="#"
    target="blank"
  >
    <i
      className="icon icon-sm ti-icon"
    />
  </a>
</SocialLink>
`;

exports[`SocialLink should render without throwing an error 1`] = `
<a
  href="#"
  target="blank"
>
  <i
    className="icon icon-sm ti-icon"
  />
</a>
`;
